content/blog/2022/2022-02-15-getting-started-with-spring-webflux.md

```diff
@@ -0,0 +1,573 @@
+---
+title: "Getting Started with Spring WebFlux"
+categories: ["Spring"]
+date: 2022-02-15 00:00:00 +1100 
+modified: 2022-02-15 00:00:00 +1100
+authors: [arpendu]
+excerpt: "A comprehensive entry level guide to understand Reactive Streams and Spring WebFlux. A deep-dive to understand the Reactive Programming paradigm and their APIs. We will also get started with Spring WebFlux and its components."
+image: images/stock/0119-webflux-1200x628.jpg 
+url: getting-started-with-spring-webflux
+---
+
+
+
+Most of the applications built in traditional and conventional world, deal with *blocking* calls or in other words, known as *synchronous* calls. This means that if we want to access a particular resource in a system with most of the threads being busy, then the application would block the new one or wait until the previous threads complete processing its requests. So as we had embarked into the world of reality dealing with the *Big Data* , we have realized that we need to process this data with immense speed and agility. That’s when the software developers realized that they would need some kind of multi-threaded environment that handles asynchronous and non-blocking calls.
```

Comment:
I already said that you're writing is very good. Sometimes, you could simplify it a bit, though,

For example here, I replaced

> Most of the applications built in traditional and conventional world ...

with

> Most traditional applications ...

I will edit some of those places during my review, nothing to do for you. Just as an FYI for your next article :)